### PR TITLE
feat(dms/kafka): support the cross-vpc modification

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Support advertised IP modification of cross-VPC.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. support the cross-vpc modification
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dms' TESTARGS='-run=TestAccKafkaInstance_newFormat'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms -v -run=TestAccKafkaInstance_newFormat -timeout 360m -parallel 4
=== RUN   TestAccKafkaInstance_newFormat
=== PAUSE TestAccKafkaInstance_newFormat
=== CONT  TestAccKafkaInstance_newFormat
--- PASS: TestAccKafkaInstance_newFormat (1322.45s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       1322.604s
```

### CODE REVIEW

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/dms' TESTARGS='-run=TestAccKafkaInstance_newFormat'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms -v -run=TestAccKafkaInstance_newFormat -timeout 360m -parallel 4
=== RUN   TestAccKafkaInstance_newFormat
=== PAUSE TestAccKafkaInstance_newFormat
=== CONT  TestAccKafkaInstance_newFormat
--- PASS: TestAccKafkaInstance_newFormat (1257.52s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       1257.622s
```

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go [hunk 1]
```diff
@@ -321,11 +325,85 @@ func getKafkaProductDetail(config *config.Config, d *schema.ResourceData) (*prod
...
+				continue
+			}
+		}
+		contentMap[lisIp] = lisIp
+	}
+
+	opts := instances.CrossVpcUpdateOpts{
+		Contents: contentMap,
+	}
+	result, err := instances.UpdateCrossVpc(client, d.Id(), opts)
```
Comment (ShiChangkuo) [inline huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go:368]: please check the API is available in all regions of huaweicloud and joined-operation clouds.

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/dms' TESTARGS='-run=TestAccKafkaInstance_newFormat'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms -v -run=TestAccKafkaInstance_newFormat -timeout 360m -parallel 4
=== RUN   TestAccKafkaInstance_newFormat
=== PAUSE TestAccKafkaInstance_newFormat
=== CONT  TestAccKafkaInstance_newFormat
--- PASS: TestAccKafkaInstance_newFormat (1335.17s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       1335.307s
```

**Pull request merged**